packages/mobile/src/merchantPayment/MerchantPaymentScreen.tsx

```diff
@@ -0,0 +1,286 @@
+import ContactCircle from 'src/components/ContactCircle'
+import FeeDrawer from 'src/components/FeeDrawer'
+import TokenDisplay from 'src/components/TokenDisplay'
+import TokenTotalLineItem from 'src/components/TokenTotalLineItem'
+import { estimateFee, FeeType } from 'src/fees/reducer'
+import { feeEstimatesSelector } from 'src/fees/selectors'
+import { UNLOCK_DURATION } from 'src/geth/consts'
+import { useCurrencyToLocalAmount } from 'src/localCurrency/hooks'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { navigateBack, navigateHome } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { getPassword, retrieveOrGeneratePepper } from 'src/pincode/authentication'
+import { useTokenInfoBySymbol } from 'src/tokens/hooks'
+import { fetchTokenBalances } from 'src/tokens/reducer'
+import { Currency } from 'src/utils/currencies'
+import { divideByWei } from 'src/utils/formatting'
+import { getContractKitAsync } from 'src/web3/contracts'
+import { isDekRegisteredSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.MerchantPayment>
+
+function MerchantPaymentScreen({ route }: Props) {
+  const { params: routeParams } = route
+  const { t } = useTranslation()
+  const e164PhoneNumber = useSelector(e164NumberSelector)
+  const dispatch = useDispatch() as (...args: unknown[]) => void
+  const [sdkCharge, setSdkCharge] = useState<Charge | null>(null)
+  const [business, setBusiness] = useState<BusinessData | null>(null)
+  const isLoading = !business
+  const [amount, setAmount] = useState(new BigNumber(0))
+  const [submitting, setSubmitting] = useState(false)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const feeEstimates = useSelector(feeEstimatesSelector)
+  const tokenInfo = useTokenInfoBySymbol(Currency.Dollar)
+  const tokenAddress = tokenInfo?.address!
+  const feeEstimate = feeEstimates[tokenAddress]?.[FeeType.SEND]
+  const storedDekFee = feeEstimates[tokenAddress]?.[FeeType.REGISTER_DEK]
+  const localToFeeExchangeRate = useCurrencyToLocalAmount(new BigNumber(1), Currency.Dollar)
+  const isDekRegistered = useSelector(isDekRegisteredSelector) ?? false
+  const navigation = useNavigation()
+
+  const initCharge = useCallback(
+    async (kit: ContractKit) => {
+      if (!kit) {
+        throw new Error('Missing kit')
+      }
+
+      const chainHandler = new ContractKitTransactionHandler(kit)
+      const charge = new Charge(routeParams.apiBase, routeParams.referenceId, chainHandler, false)
+      try {
+        const info = await charge.getInfo()
+        setSdkCharge(charge)
+        setAmount(divideByWei(new BigNumber(info.action.amount)).decimalPlaces(2))
+        setBusiness(info.receiver.businessData)
+      } catch (e: unknown) {
+        dispatch(showError(t('merchantPaymentSetup')))
+        navigateBack()
```

Comment:
There's an edge case here too if the user has already left the screen, this shouldn't be called.